_posts/2024-10-08-Starting-Git.md

```diff
@@ -0,0 +1,500 @@
+
+
+
+### Push changes to a remote repo
+
+If you have a remote repo, like on GitHub, ou need to push your commits to that repo so that people can actually see your changes 
+
+- command:
+    - git push origin "branch name"
+
+[git push origin main] -> Pushes the changes from your local repository to the "main" branch of the remote repo
+
+
+
+### Pull changes from a remote repo
+
+If someone else made a change to a remote repo, you can pull those changes to update your local repo
+
+- command:
+    - git pull origin "branch-name"
+
+
+
+### Create a new branch
+
+Branches allow you to work on seperate features, bug fixes or experiments without affecting the main codebase
+
+- Command:
+    - git branch <branch-name>
+
+
+
+### Switch between branches
+
+After creating a branch, you can switch to it to start working on it  
+
+Command:
+    - git checkout "branch-name"
+
+
+
+### Merge branches
+
+Once you're done with a branch, you can merge it back in to the main branch or any other branch 
+
+- Command:
+    - git merge <branch-name>
+
+
+
+### Discard unchanged changes
+
+If you've made any changes that you don't want to keep, you can just discard them
+
+- Command:
+    - git checkout -- <filename>
+
+
+
+### Remove files from staging area
+
+If you've staged a file you'd no longer like to commit, you can remove it from the stagihng area without deleting the file
+
+
+- Command:
+    - git reset "filename"
+
+
+From what I can tell, these are the main ones required to manage everyday tasks in git.
+
+
+___
+
+
+# Step 4: Understanding Branching and merging
+
+Branching and merging are two of the most powerful features of git..... apparently. They allow you to work on different features, bug fixes or experiments without affecting the main codebase. This will also help give a actual reasons why you'd need the above stuff.
+
+
+Let's start with the basics.... What is a branch?
+
+A branch in git represents an independant line of development. Like creating a new timeline that has no effect on the main project
+- This allows you to:
+    - Work on new features without disturbing the main branch
+     - Collaborate on the mainbase without conflicting with other peoples work
+      - Safely experiment with new ideas and roll back if necessary
+
+each git repository starts with a default branch called "main"
+
+
+Now I would tell you the command as we go, but if you don't know what code I'm explaining, it means you skipped over that part, and I do not reward bad behaviour....gave that list for a reason.......
+
+To add an additional branch to work outside of main, the above command to add a branch will do that. 
+
+Now that you understand why having multiple branches helps a lot with production, you should probably also switch between each branch, right? 
+The code to do that is above as well ;)
+
+
+Seriously now, we'll go in to branch specific features I never cover yet:
+
+### View all branches
+
+- Self-explainatory
+    - git branch

```

Comment:
```suggestion
    - `$ git branch`

```

Use single backticks ` to denote inline codeblocks.


`like this`

Start all terminal commands with $ to denote it is a command in the cli

can also do triple backticks to make full codeblock:

```python
for i in range(2):
  print('cool')
```